### PR TITLE
Improving PLD

### PR DESCRIPTION
**Pull request by nksaunders**
With the help of @dfm, this PR makes a few changes to speed up the `PLDCorrector` class.

 - Remove deprecated dependencies
 - Remove `np.diag` assignment
 - Consolidate redundant calculation of `sigma`
 - replace `np.linalg.solve` with `scipy.linalg.cho_solve`

### CODE REVIEW

**Comment by barentsen** [comment]
@nksaunders I fixed the failing unit test over in #471.  It should go away if you `git rebase master`, which should work without conflict!

**Comment by barentsen** [comment]
This looks good!  I don't see anything that concerns me.  :+1:

**Comment by nksaunders** [comment]
@barentsen, could you take one more look at this and merge if it looks good? We have users eager to try it! 🎉

File: appveyor.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ install:
   - "ECHO %PATH%"
   - where conda
   - conda config --set always_yes yes --set changeps1 no
-  - "conda create -q -n test-environment python=%PYTHON_VERSION% numpy astropy scipy matplotlib requests tqdm pandas bokeh ipython scikit-learn pybind11 pytest pytest-cov pytest-remotedata"
+  - "conda create -q -n test-environment python=%PYTHON_VERSION% numpy astropy scipy matplotlib requests tqdm pandas bokeh ipython pybind11 pytest pytest-cov pytest-remotedata"
```
Comment (barentsen) [inline appveyor.yml]: This is only for testing, but do we want `fbpca` and `celerite` here, or did it break in the past?

File: setup.py [hunk 2]
```diff
@@ -22,10 +22,10 @@
 # 3. What dependencies are required for optional features?
 # `BoxLeastSquaresPeriodogram` requires astropy>=3.1.
 # `interact()` requires bokeh>=1.0, ipython.
-# `PLDCorrector` requires scikit-learn, pybind11, celerite.
+# `PLDCorrector` requires pybind11, celerite.
```
Comment (barentsen) [inline setup.py]:
```suggestion
# `PLDCorrector` requires pybind11, celerite, fbpca.
```

File: lightkurve/correctors.py [hunk 3]
```diff
@@ -618,15 +618,20 @@ def correct(self, aperture_mask=None, cadence_mask=None, gp_timescale=30,
...
+            this to False to speed up the correction (at the cost of precision),
+            or if you suspect the presence of systematic noise at long timescales.
+        pld_order : int
+            The order of Pixel Level De-correlation to be performed. First order
+            (`n=1`) uses only the pixel fluxes to construct the design matrix.
+            Higher order populates the design matrix with columns constructed
+            from the products of pixel fluxes.
+        n_pca_terms : int
+            Number of terms added to the design matrix from each order of PLD
+            when performing Principle Component Analysis for models higher than
```
Comment (barentsen) [inline lightkurve/correctors.py]:
```suggestion
            when performing Principal Component Analysis for models higher than
```

File: lightkurve/correctors.py [hunk 4]
```diff
@@ -673,38 +669,52 @@ def correct(self, aperture_mask=None, cadence_mask=None, gp_timescale=30,
...
+        pld_flux = flux_crop[:, aperture_crop]
+        f1 = np.reshape(pld_flux, (len(pld_flux), -1))
+        X1 = f1 / np.sum(pld_flux, axis=-1)[:, None]
+
+        # higher order PLD design matrices
+        X_sections = [np.ones((len(flux_crop), 1)), X1]
+        for i in range(2, pld_order+1):
+            f2 = np.product(list(multichoose(X1.T, pld_order)), axis=1).T
+            try:
+                # optionally dependency for PCA
```
Comment (barentsen) [inline lightkurve/correctors.py]:
```suggestion
                # We use an optional dependency for very fast PCA (fbpca).
                # If the import fails we will fall back on using the slower `np.linalg.svd`
```

**Pull request merged**